raiden-ts/tests/unit/epics/monitor.spec.ts

```diff
@@ -357,6 +369,268 @@ describe('monitorRequestEpic', () => {
+
+    const partnerBP = getChannel(raiden, partner).partner.balanceProof;
+
+    expect(raiden.output).toContainEqual(
+      messageGlobalSend(
+        {
+          message: {
+            type: MessageType.MONITOR_REQUEST,
+            balance_proof: {
+              chain_id: partnerBP.chainId,
+              token_network_address: tokenNetwork,
+              channel_identifier: partnerBP.channelId,
+              nonce: partnerBP.nonce,
+              balance_hash: createBalanceHash(partnerBP),
+              additional_hash: partnerBP.additionalHash,
+              signature: partnerBP.signature,
+            },
+            non_closing_participant: raiden.address,
+            non_closing_signature: expect.any(String),
+            monitoring_service_contract_address: expect.any(String),
+            reward_amount: raiden.config.monitoringReward!,
+            signature: expect.any(String),
+          },
+        },
+        { roomName: expect.stringMatching(/_monitoring$/) },
+      ),
+    );
+  });
+
+  test('ignore: not enough udcBalance', async () => {
+    expect.assertions(2);
+
+    const [raiden, partner] = await makeRaidens(2);
+    const { userDepositContract } = raiden.deps;
+    const monitoringReward = bigNumberify(5) as UInt<32>;
+    raiden.store.dispatch(
+      raidenConfigUpdate({
+        httpTimeout: 30,
+        monitoringReward,
+        caps: {
+          [Capabilities.NO_DELIVERY]: true,
+          [Capabilities.NO_MEDIATE]: true,
+          // 'noReceive' should be auto-disabled by 'getCaps$'
+        },
+        // rate=WeiPerEther == 1:1 to SVT
+        rateToSvt: { [token]: WeiPerEther as UInt<32> },
+      }),
+    );
+
+    userDepositContract.functions.effectiveBalance.mockResolvedValue(monitoringReward.sub(1));
+    await waitBlock();
+
+    expect(raiden.output).toContainEqual(
+      udcDeposit.success(undefined, { totalDeposit: monitoringReward.sub(1) as UInt<32> }),
+    );
+
+    await ensureChannelIsDeposited([raiden, partner], deposit);
+    await ensureChannelIsDeposited([partner, raiden], deposit);
+    await ensureTransferUnlocked([partner, raiden], amount);
+    await waitBlock();
+
+    expect(raiden.output).not.toContainEqual(
+      messageGlobalSend(
+        { message: expect.objectContaining({ type: MessageType.MONITOR_REQUEST }) },
+        expect.anything(),
+      ),
+    );
+  });
+
+  test('ignore: config.monitoringReward unset', async () => {
+    expect.assertions(1);
+    const [raiden, partner] = await makeRaidens(2);
+    raiden.store.dispatch(raidenConfigUpdate({ monitoringReward: null }));
+
+    await ensureChannelIsDeposited([raiden, partner], deposit);
+    await ensureChannelIsDeposited([partner, raiden], deposit);
+    await ensureTransferUnlocked([partner, raiden], amount);
+    await waitBlock();
+
+    expect(raiden.output).not.toContainEqual(
+      messageGlobalSend(
+        { message: expect.objectContaining({ type: MessageType.MONITOR_REQUEST }) },
```

Comment:
Very nice! Broad `.not.toContainEqual`, and strict whole message checks when it does